src/braket/experimental/autoqasm/instructions/instructions.py

```diff
@@ -31,8 +42,43 @@ def _qubit_instruction(
     program_conversion_context.register_gate(name)
     program_conversion_context.register_args(args)
     program_mode = aq_program.ProgramMode.UNITARY if is_unitary else aq_program.ProgramMode.NONE
+    pos_control, neg_control = _get_pos_neg_control(control, control_state)
     oqpy_program = program_conversion_context.get_oqpy_program(mode=program_mode)
-    oqpy_program.gate([_qubit(q) for q in qubits], name, *args)
+    oqpy_program.gate(
+        [_qubit(q) for q in qubits],
+        name,
+        *args,
+        control=pos_control,
+        neg_control=neg_control,
+        exp=power,
+    )
+
+
+def _get_pos_neg_control(
+    control: QubitIdentifierType | Iterable[QubitIdentifierType] | None = None,
+    control_state: str | None = None,
+) -> tuple[Iterable[oqpy.Qubit], Iterable[oqpy.Qubit]]:
+    if control is None and control_state is not None:
+        raise ValueError(control_state, "control_state provided without control qubits")
+
+    if control is None:
+        return None, None
+
+    if isinstance(control, str) or not isinstance(control, Iterable):
```

Comment:
should we use [is_qubit_identifier_type](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/d82bac562075ab7878e17744f912a78f4f00fff9/src/braket/experimental/autoqasm/types/types.py#L63) instead of `str` ? The `str` is for physical qubit but is a bit out of blue here